### PR TITLE
Recursive support for helper function array_urlencode and array_toque…

### PR DESCRIPTION
**Pull request by jjonas-gcs**
Please read our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR. 

Fixes # 

## Additional info  

…rystring

### CODE REVIEW

File: pimcore/config/helper.php [hunk 1]
```diff
@@ -126,53 +126,54 @@ function object2array($node)
...
-            }
-        } else {
-            $out .= urlencode($name).'=';
-            $out .= urlencode($value).'&';
-        }
-    }
-
-    return substr($out, 0, -1); //trim the last & }
+    
+    return str_replace(['%5B', '%5D'], ['[', ']'], http_build_query($args));
```
Comment (brusch) [inline pimcore/config/helper.php]: This solution can be problematic when having the following source array: 

```php
        $foo = [
            "one" => [
                "level2" => [
                    "value" => "[myValue]"
                ]
            ]
        ];
```

so I guess it's better to use also array_toquerystringHelper but with an additional parameter for urlencode. 
What do you think?

Comment (jjonas-gcs) [inline pimcore/config/helper.php]: Maybe the best solution is just call the http_build_query function without the str_replace?

Do you know in which cases pimcore needs the returned string of this function or is the method only used for output?

I open the following links in chrome and firefox:
* /?one%5Blevel2%5D%5Bvalue%5D=%5BmyValue%5D
* /?one[level2][value]=[myValue]
and i got the same result for var_dump( $_GET ). I think it does not matter if you call a link with [ and ] or with %5B and %5D.

I would prefer to delete the str_replace call before http_build_query, but if u want i will change it to the array_toquerystringHelper method.

It is your choice just tell me what to do and I will do it :)

Comment (brusch) [inline pimcore/config/helper.php]: Good question :) 
If it doesn't matter, I'd also prefer removing the `str_replace`. 

Thanks!

Comment (jjonas-gcs) [inline pimcore/config/helper.php]: Hi brusch, 
i updated my pull request and now using only http_build_query.
Regards Jens

**Pull request merged**